Modules/CTP/src/RawDataReaderCheck.cxx

```diff
@@ -81,55 +135,89 @@ Quality RawDataReaderCheck::check(std::map<std::string, std::shared_ptr<MonitorO
           vGoodBC.push_back(i);
         }
       }
-      if (vBadBC.size() > 0)
+      if (vBadBC.size() > 0) {
         result = Quality::Bad;
-      else if (vMediumBC.size() > 0)
+      } else if (vMediumBC.size() > 0) {
         result = Quality::Medium;
-      else
+      } else {
         result = Quality::Good;
-    }
-    if (mo->getName() == "inputs" || mo->getName() == "classes") {
+      }
+    } else {
       auto* h = dynamic_cast<TH1F*>(mo->getObject());
-      if (mo->getName() == "inputs" && (!fHistInputPrevious || inputCycleCounter < 2)) {
-        fHistInputPrevious = (TH1F*)h->Clone();
-        inputCycleCounter++;
-      } else if (mo->getName() == "classes" && (!fHistClassesPrevious || classCycleCounter < 2)) {
-        fHistClassesPrevious = (TH1F*)h->Clone();
-        classCycleCounter++;
-      } else {
+      std::string moName = mo->getName();
+      if (moName.find("Ratio") != std::string::npos) {
+        relativeRates = true;
+      }
+      if (moName.find("input") != std::string::npos) {
+        inputRates = true;
+      }
+      if (relativeRates && inputRates) {
+        h->Scale(1. / h->GetBinContent(3));
+      } else if (relativeRates) {
+        h->Scale(1. / h->GetBinContent(mIndexMBclass));
+      }
+
+      if (cycleCounter > 1) { // skipping the check for first two cycles, as the first one is arbitrary long
         TH1F* fHistDifference = (TH1F*)h->Clone();
```

Comment:
I don't see anywhere that you assign this pointer elsewhere or delete it. It looks like a memory leak.